actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java

```diff
@@ -29,26 +29,28 @@ public void onReceive(Request request) throws Throwable {
     }
   }
 
-  private void sendOTP(Request request) {
+  private void sendOTP(Request request) throws Exception {
     String type = (String) request.getRequest().get(JsonKey.TYPE);
     String key = (String) request.getRequest().get(JsonKey.KEY);
     String otp = (String) request.getRequest().get(JsonKey.OTP);
+    String template = (String) request.getRequest().get(JsonKey.TEMPLATEID);
     if (JsonKey.EMAIL.equalsIgnoreCase(type) || JsonKey.PREV_USED_EMAIL.equalsIgnoreCase(type)|| JsonKey.RECOVERY_EMAIL.equalsIgnoreCase(type)) {
       String userId = (String) request.get(JsonKey.USER_ID);
       ProjectLogger.log(
           "SendOTPActor:sendOTP userId found for send otp " + userId, LoggerEnum.INFO.name());
-      sendOTPViaEmail(key, otp, userId);
+      sendOTPViaEmail(key, otp, userId, template);
     } else if (JsonKey.PHONE.equalsIgnoreCase(type)
         || JsonKey.PREV_USED_PHONE.equalsIgnoreCase(type) || JsonKey.RECOVERY_PHONE.equalsIgnoreCase(type)) {
-      sendOTPViaSMS(key, otp);
+      sendOTPViaSMS(key, otp, template);
     }
   }
 
-  private void sendOTPViaEmail(String key, String otp, String otpType) {
+  private void sendOTPViaEmail(String key, String otp, String otpType, String template) {
     Map<String, Object> emailTemplateMap = new HashMap<>();
     emailTemplateMap.put(JsonKey.EMAIL, key);
     emailTemplateMap.put(JsonKey.OTP, otp);
     emailTemplateMap.put(JsonKey.OTP_EXPIRATION_IN_MINUTES, OTPUtil.getOTPExpirationInMinutes());
+    emailTemplateMap.put(JsonKey.TEMPLATEID, template);

```

Comment:
If provided template is incorrect then it should throw data error.

Reply:
Taken care in controller.